connections/ethereum/gsn/gsn.go

```diff
@@ -0,0 +1,86 @@
+package gsn
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+	"path/filepath"
+	"time"
+)
+
+type RPCError struct {
+	Code    int         `json:"code"`
+	Message string      `json:"message"`
+	Data    interface{} `json:"data,omitempty"`
+}
+
+type Response struct {
+	Error  *RPCError       `json:"error"`
+	ID     int             `json:"id"`
+	Result json.RawMessage `json:"result,omitempty"`
+}
+
+type gsnResponse struct {
+	Fast          int64       `json:"fast"`
+	Fastest       int64       `json:"fastest"`
+	SafeLow       int64       `json:"safeLow"`
+	Average       int64       `json:"average"`
+	Block_time    float32     `json:"block_time"`
+	BlockNum      int64       `json:"blockNum"`
+	Speed         float32     `json:"speed"`
+	SafeLowWait   float32     `json:"safeLowWait"`
+	AvgWait       float32     `json:"avgWait"`
+	FastWait      float32     `json:"fastWait"`
+	FastestWait   float32     `json:"fastestWait"`
+	GasPriceRange interface{} `json:"gasPriceRange"`
+}
+
+// callGSN will call the Gas Station Network and request the gas prices on the Ethereum network
+func CallGSN(apiKey, speed string) (error, *big.Int) {
```

Comment:
There is no tests for this function.
Also it is impossible to unittest it since Client that makes external request in inside this function. 
I think this function should accept some kind of  Client interface or be part of some struct that holds Client as interface, so this Client could be substituted with mocked implementation in sake of unit testing